### ISSUE TITLE
Scream not creating second IVSHMEM device on Win10

### ISSUE THREAD

**Issue by danipoak** [duncanthrax/scream#101]
Ultimately I'm trying to pass through my audio from Win10 to Gentoo using scream. Following the basic README instructions from the Windows download I right clicked, and ran as administrator. Everything seemed to proceed as normal, no errors. I then opened up the device manager to set the IVSHMEM driver. I already have Looking Glass installed so there is one IVSHMEM device for that with the driver installed, but there is no second device for me to set the driver for. Nor is there a `PCI standard RAM Controller` . You can see from the screen capture that scream is the selected output but there is no IVSHMEM device for scream. Wondering if this is perhaps expected behavior I went ahead and installed a receiver on the Gentoo side. I'm using Pulseaudio. Using the basic instructions I did as follows:
```
$ mkdir build && cd build
$ cmake ..
$ make
```
I then set a symbolic link to /usr/bin
` sudo touch /dev/shm/scream && sudo chown steve /dev/shm/scream && sudo chmod 660 /dev/shm/scream`
Using QEMU I added the recommended lines, adjusting for my scenario. Note that I left the QEMU entries in for Looking Glass as well in case there is conflict.
```
    -device ivshmem-plain,memdev=ivshmem,bus=pci.0 \
    -object memory-backend-file,id=ivshmem,share=on,mem-path=/dev/shm/looking-glass,size=32M \
    -device ivshmem-plain,memdev=ivshmem,bus=pci.1 \
    -object memory-backend-file,id=ivshmem,share=on,mem-path=/dev/shm/scream,size=2M \
```
Then executing the receiver as follows:
```
$ scream -m /dev/shm/scream 
Failed to map the shared memory file: /dev/shm/scream
```
So I guess the question is: is there a bug somewhere that isn't creating the second IVSHMEM instance on Windows, or is my process all wrong and problem is between char and keyboard?
 
![Capture](https://user-images.githubusercontent.com/17608130/86069464-e7a4c100-ba47-11ea-8991-6d7b0923ef1b.PNG)

**Comment by martinellimarco** [comment]
I think the problem is that both your IVSHMEM devices share the same memdev id.
Try replacing one like this:

```
    -device ivshmem-plain,memdev=ivshmem-scream,bus=pci.1 \
    -object memory-backend-file,id=ivshmem-scream,share=on,mem-path=/dev/shm/scream,size=2M \
```

**Comment by danipoak** [comment]
I was hoping it would be so simple! I modified my QEMU script like you suggested:
```
    -device ivshmem-plain,memdev=ivshmem,bus=pci.0 \
    -object memory-backend-file,id=ivshmem,share=on,mem-path=/dev/shm/looking-glass,size=32M \
    -device ivshmem-plain,memdev=ivshmem-scream,bus=pci.1 \
    -object memory-backend-file,id=ivshmem-scream,share=on,mem-path=/dev/shm/scream,size=2M \
```
Result is the same:
```
$ scream -m /dev/shm/scream 
Failed to map the shared memory file: /dev/shm/scream
```
I noticed that after a restart the shared memory file I created using these commands:
`sudo touch /dev/shm/scream && sudo chown steve /dev/shm/scream && sudo chmod 660 /dev/shm/scream`
Goes away and I had to create it again. Should scream be creating the shared memory file automatically when I call QEMU? There aren't any instructions saying to do so but I remembered from the Looking Glass instructions that you should. 

Should there be a second IVSHMEM device? One for Looking Glass and one for Scream? Or does Scream share the IVSHMEM device in Win10?

**Comment by martinellimarco** [comment]
In Windows you should see 2 IVSHMEM devices.

QEMU should create the file in /dev/shm/ on start. Don't manually create it but make sure you have read permission on it.

**Comment by danipoak** [comment]
Okay it works! I'm not exactly sure where it went wrong but I have a good guess. I deleted everything and started from the beginning. I think the confusing bit of the top level Readme is:
`If you use other IVSHMEM devices, we recommend to use the same domain and bus, just varying the slot numbers. `
So looking at the recommended lines to QEMU, they differ slightly from those added for looking glass:
Scream:
```
    -device ivshmem-plain,memdev=ivshmem_scream \
    -object memory-backend-file,id=ivshmem_scream,share=on,mem-path=/dev/shm/scream-ivshmem,size=2M \
```
Looking-Glass:
```
    -device ivshmem-plain,memdev=ivshmem,bus=pci.0 \
    -object memory-backend-file,id=ivshmem,share=on,mem-path=/dev/shm/looking-glass,size=32M \
```
The difference being no Bus specified on Scream while `bus=pci.0` is specified on Looking-Glass. I interpreted the readme as needing to add `bus=pci.1` to my QEMU call but in reality it isn't needed. Re-reading things I *think* the line `If you use other IVSHMEM devices, we recommend to use the same domain and bus, just varying the slot numbers. ` is referring only to libvrt/virt-manager and not to calling QEMU directly.
